### PR TITLE
+ code snippet component

### PR DESCRIPTION
**Pull request by dominikvoda**

### CODE REVIEW

File: src-php/Typography/CodeSnippetTest.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    public function testCodeSnippetIsRendered(): void
+    {
+        $code = <<<EOT
+<script type="text/javascript">
+        alert('Hello boys & girls!')
+    </script>
+EOT;
+
+        $escapedCode = str_replace(['&', '<'], ['&amp;', '&lt;'], $code);
```
Comment (janveselynet) [inline src-php/Typography/CodeSnippetTest.php]: Why to escape here?

File: src-php/Typography/CodeSnippet.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        return '
+        <pre style="
+                white-space: pre-wrap;
+                box-sizing: border-box;
+                background-color: #f4f4f4;
+                width: 100%;
+                margin-top: 20px;
+                padding: 14px;
+            ">
+            ' . $this->code . '
```
Comment (janveselynet) [inline src-php/Typography/CodeSnippet.php]: I would add escape

File: src-php/Typography/CodeSnippet.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    public function __construct(string $code)
+    {
+        $this->code = $code;
+    }
+
+
+    public function render(): string
+    {
+        $escapedCode = str_replace(['&', '<'], ['&amp;', '&lt;'], $this->code);
```
Comment (michalbrauner) [inline src-php/Typography/CodeSnippet.php]: why not to use `htmlspecialchars` ?

File: src-php/Typography/__snapshots__/codeSnippet.html [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    white-space: pre-wrap;
+    box-sizing: border-box;
+    background-color: #f4f4f4;
+    width: 100%;
+    margin-top: 20px;
+    padding: 14px;
+">
+    &lt;script type="text/javascript">
+        alert('Hello boys &amp; girls!')
+    &lt;/script>
```
Comment (michalbrauner) [inline src-php/Typography/__snapshots__/codeSnippet.html]: there is missing `>`

File: src-php/Typography/CodeSnippet.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        return '
+        <pre style="
+                white-space: pre-wrap;
+                box-sizing: border-box;
+                background-color: #f4f4f4;
+                width: 100%;
+                margin-top: 20px;
+                padding: 14px;
+            ">
+            ' . htmlspecialchars($this->code) . '
```
Comment (janveselynet) [inline src-php/Typography/CodeSnippet.php]: Use \BrandEmbassy\Components\StringEscaper

**Pull request merged**